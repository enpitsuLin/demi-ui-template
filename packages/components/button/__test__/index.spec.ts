import DButton from '../src/button.vue'
import { mount } from '@vue/test-utils'

describe('Button', () => {
  it('create', () => {
    const wrapper = mount(DButton, { slots: { default: 'default slot' } })

    expect(wrapper.text()).toContain('default slot')

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('type property work', () => {
    /**
     * @desc use deprecated `propsData` compatibility for `@vue/test-utils@1` & `@vue/test-utils@2`
     * @todo create package to compatible w/ both v1 & v2
     */
    const wrapper = mount(DButton, {
      propsData: {
        type: 'primary',
      },
      slots: { default: 'default slot' },
    })

    expect(wrapper.classes()).toContain('d-button--primary')
  })
})
