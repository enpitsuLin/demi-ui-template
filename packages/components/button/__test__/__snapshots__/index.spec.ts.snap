// Vitest Snapshot v1

exports[`mount component 1`] = `"<button class=\\"button\\"><span>count 0</span></button>"`;
